src/main/kotlin/note/application/NoteUpdater.kt

```diff
@@ -0,0 +1,16 @@
+package note.application
+
+import note.domain.Note
+import note.domain.NoteIdentifier
+import note.domain.NoteIdentifierGenerator
+import note.domain.NoteRepository
+
+class ReminderUpdater(
+    private val repository: NoteRepository,
+    private val identifierGenerator: NoteIdentifierGenerator,
+) {
+    fun update(identifier: NoteIdentifier, newTitle: String, newDescription: String?): Note {
+        NoteDeleter(repository).delete(identifier)
```

Comment:
What happens if the noteIdentifier doesn't exist on the NoteRepository? We should throw an exception.

Anyway, I was thinking about comparing the primitives (title and description) of the current note and, if they are equal, return the current note and don't save anything to the repository. For this implementation, maybe we should refactor and pass to the update and create use cases a "NotePrimitive(newTitle: string, newDescription:string)" and a note could have a .toPrimitives() method that converts the object to his primitives. This use case then would be: 

```kotlin
fun update(identifier: NoteIdentifier, notePrimitives: NotePrimitives):
    val currentNote = NoteDeleter(repository).find(identifier) ?: throw NoSuchIdentifierException()
    if (primitives == note.toPrimitives()) return currentNote
    NoteDeleter(repository).delete(identifier)
    NoteCreator(repository).create(identifier, primitives).
```

Reply:
The use case of deleting a reminder aleeady contemplates the case where the identifier doesn't exist (and throws an exception); so I think there's no need to implement that functionality twice. 
But I do like what you said about comparing the primitives.

Reply:
Yes, my bad, I have to stop reviewing code at night.  If we do the Primitive change, we should change the create note use case too.

Reply:
You forgot to rename the class to NoteUpdater 👉👈 ...

Reply:
The renaming has already been done, next time put it in a different thread please 🌸

When it comes to the primitive change, I've thought some things about it:

- There's no need to implement the whole "identifier doesn't exist -> throw exception" case, UNLESS ...
- If we want to implement the primitive process, the use case should go as you stated:
    - Search the identifier
        - If it doesn't exist -> throw exception
        - If it does exist -> recover note
    - When we have the note:
        - If the note is the same (primitive-likewise) -> return the same note
        - If the note is not the same note:
            - Actually delete the note
            - Create a new note with the new info
The problem is that **the deletion process looks for the note, throws exception if necessary, and then deletes it if exists**
We would have to intercept this behaviour in the middle, or otherwise fetch the note before and act in consequence.
There's also the option of re-instantiating the "same" note again.

Given this, what approach would you guys take?

Reply:
If it is not the same note, we have to follow this steps:
1) Delete the current note from the repository (we have the note identifier from the fetched note!)
2) Create a new note inside the repository with the new changes.


I don't see any problem on that behavior